src/number_theory/number_field.lean

```diff
@@ -142,3 +144,89 @@ noncomputable def ring_of_integers_equiv : ring_of_integers ℚ ≃+* ℤ :=
+
+local attribute [-instance] algebra_rat
+
+/-- The quotient of `ℚ[X]` by the ideal generated by an irreducible polynomial of `ℚ[X]`
+is a number field. -/
+instance {f : ℚ[X]} [hf : irreducible f] : number_field (adjoin_root f) :=
+{ to_char_zero := char_zero_of_injective_algebra_map (algebra_map ℚ _).injective,
+  to_finite_dimensional := begin
+   let := (adjoin_root.power_basis (irreducible.ne_zero hf : f ≠ 0)),
+   convert power_basis.finite_dimensional this,
+   haveI : subsingleton (algebra ℚ (adjoin_root f)) := algebra_rat_subsingleton,
+   exact subsingleton.elim _ _,
+  end }
+
+end
+
+end adjoin_root
+
+namespace number_field.embeddings
+
+section number_field
+
+open set finite_dimensional polynomial
+
+variables {K L : Type*} [field K] [field L]
+variables [number_field K] [number_field L]  (x : K)
+
+variables {A : Type*} [field A] [char_zero A]
+
+/-- There are finitely many embeddings of a number field. -/
+noncomputable instance : fintype (K →+* A) := fintype.of_equiv (K →ₐ[ℚ] A)
+ring_hom.equiv_rat_alg_hom.symm
+
+variables [is_alg_closed A]
+
+/-- The number of embeddings of a number field is its finrank. -/
+lemma card_embeddings : fintype.card (K →+* A) = finrank ℚ K :=
+by rw [fintype.of_equiv_card ring_hom.equiv_rat_alg_hom.symm, alg_hom.card]
+
+/-- For `x ∈ K`, with `K` a number field, the images of `x` by the embeddings of `K` are exactly
+the roots of the minimal polynomial of `x` over `ℚ` -/
+lemma eq_roots : range (λ φ : K →+* A, φ x) = (minpoly ℚ x).root_set A :=
+begin
+  have hx : is_integral ℚ x, { exact is_separable.is_integral ℚ x },
+  ext a, split,
+  { rintro ⟨φ, hφ⟩,
+    rw [mem_root_set_iff, ←hφ],
+    let ψ := ring_hom.equiv_rat_alg_hom φ,
+    show (aeval (ψ x)) (minpoly ℚ x) = 0,
+    { rw aeval_alg_hom_apply ψ x (minpoly ℚ x),
+      simp only [minpoly.aeval, map_zero] },
+    exact minpoly.ne_zero hx, },
+  { intro ha,
+    let Qx := adjoin_root (minpoly ℚ x),
+    haveI : irreducible (minpoly ℚ x), { exact minpoly.irreducible hx },
+    have hK : (aeval x) (minpoly ℚ x) = 0, { exact minpoly.aeval _ _, },
+    have hA : (aeval a) (minpoly ℚ x) = 0,
+    { rw [aeval_def, ←eval_map, ←mem_root_set_iff'],
+      exact ha,
+      refine polynomial.monic.ne_zero _,
+      exact polynomial.monic.map (algebra_map ℚ A) (minpoly.monic hx), },
+    let ψ : Qx →+* A := adjoin_root.lift (algebra_map ℚ A) a hA,
+    letI : algebra Qx A, { exact ring_hom.to_algebra ψ },
+    letI : algebra Qx K,
+    { exact ring_hom.to_algebra (adjoin_root.lift (algebra_map ℚ K) x hK), },
+    let φ₀ : K →ₐ[Qx] A := is_alg_closed.lift _,
+    let φ := φ₀.to_ring_hom,
+    use φ,
+    rw (_ : x = (algebra_map Qx K) (adjoin_root.root (minpoly ℚ x))),
+    rw (_ : a = ψ (adjoin_root.root (minpoly ℚ x))),
+    refine alg_hom.commutes _ _,
+    exact (adjoin_root.lift_root hA).symm,
+    exact (adjoin_root.lift_root hK).symm,
+    refine algebra.is_algebraic_of_larger_base ℚ Qx _,
+    exact number_field.is_algebraic _, },
```

Comment:
This part is a little hard to follow, since you introduce new goals. Maybe it can be indented better, for readability.